packages/application-shell/src/components/project-switcher/project-switcher.js

```diff
@@ -96,108 +99,91 @@ const mapProjectsToOptions = defaultMemoize(projects =>
   }))
 );
 
-export class ProjectSwitcher extends React.PureComponent {
-  static displayName = 'ProjectSwitcher';
+const redirectTo = targetUrl => window.location.replace(targetUrl);
 
-  static propTypes = {
-    projectKey: PropTypes.string.isRequired,
-    total: PropTypes.number.isRequired,
-    // injectIntl
-    intl: PropTypes.shape({
-      formatMessage: PropTypes.func.isRequired,
-    }).isRequired,
-    // graphql
-    projectsQuery: PropTypes.shape({
-      loading: PropTypes.bool.isRequired,
-      error: PropTypes.shape({
-        message: PropTypes.string.isRequired,
-      }),
-      user: PropTypes.shape({
-        projects: PropTypes.shape({
-          results: PropTypes.arrayOf(
-            PropTypes.shape({
-              key: PropTypes.string.isRequired,
-              name: PropTypes.string.isRequired,
-              suspension: PropTypes.shape({
-                isActive: PropTypes.bool.isRequired,
-              }),
-              expiry: PropTypes.shape({
-                isActive: PropTypes.bool.isRequired,
-              }),
-            })
+const ProjectSwitcher = props => {
+  const intl = useIntl();
+
+  if (props.projectsQuery.loading) return null;
+
+  return (
+    <div
+      className={styles['react-select-wrapper']}
+      data-track-component="ProjectSwitch"
+      data-track-event="click"
+    >
+      <SelectInput
+        value={props.projectKey}
+        name="project-switcher"
+        aria-labelledby="project-switcher"
+        onChange={event => {
+          const selectedProjectKey = event.target.value;
+          if (selectedProjectKey !== props.projectKey)
+            // We simply redirect to a "new" browser page, instead of using the
+            // history router. This will simplify a lot of things and avoid possible
+            // problems like e.g. resetting the store/state.
+            redirectTo(`/${selectedProjectKey}`);
+        }}
+        options={
+          props.projectsQuery.user &&
+          mapProjectsToOptions(props.projectsQuery.user.projects.results)
+        }
+        isOptionDisabled={option =>
+          option.suspension.isActive || option.expiry.isActive
```

Comment:
This was the bug, as suspended/expired projects should not be clickable.

Reply:
👍